internal/resources/resource_volume.go

```diff
@@ -279,7 +280,48 @@ func resourceMetalVolumeRead(d *schema.ResourceData, meta interface{}) (err erro
 func resourceMetalVolumeUpdate(d *schema.ResourceData, meta interface{}) (err error) {
 	defer wrapResourceError(&err, "failed to update volume")
 
-	//@TODO - or not....?
+	c, err := client.GetClientFromMetaMap(meta)
+	if err != nil {
+		return
+	}
+
+	ctx := c.GetContext()
+
+	vol, _, err := c.Client.VolumesApi.GetByID(ctx, d.Id())
+	if err != nil {
+		return
+	}
+
+	newSize, ok := d.Get(vSize).(float64)
+	if !ok {
+		return fmt.Errorf("size is not in the expected format")
+	}
+
+	vol.Capacity = int64(newSize)
+
+	_, _, err = c.Client.VolumesApi.Update(ctx, vol)
+	if err != nil {
+		return
+	}
+
+	for {
```

Comment:
Is there an opportunity to loop forever here? Maybe this is a common pattern for TF providers however I worry that there is no max loop/max timeout.

Reply:
it can happen, I was trying to follow the pattern for the async operations in TF. But it is reasonable to abort waiting after the max timeout.
@AnuBose @llabordehpe, any idea how long does typically Volume capacity update operation takes?

Reply:
@llabordehpe volmon timeout is a configurable field, and right now, it is 5mins across the 3 environments.
I guess it is OK to return an error in TF update even though the update API is still in progress ( in the case where volmon timeout is bigger and taking long to expand capacity) as the next rerun or refresh will get the updated value.